### ISSUE TITLE
Arrow Keys with disabled entries

### ISSUE THREAD

**Issue by rssteffey** [reactjs/react-autocomplete#237]
Arrow keys up/down continue to select disabled menu items (in my particular case, category headers) even though the highlight doesn't render.  Pressing enter while over these will still allow selection of the disabled items

**Comment by CMTegner** [comment]
Hi Shawn! Could you please provide a reduced test case so we can debug this problem?

There's an example of how to add disabled menu items [here](https://reactcommunity.org/react-autocomplete/custom-menu/). Maybe you can compare it to your code and see if there are any obvious differences that may cause problems.

**Comment by rssteffey** [comment]
Sorry, Christian.  I've actually inherited this project and had assumed that the "disabled" property on our menu items was the proper method.

By my understanding of that example the only "correct" way to create category headers is by rendering an additional div based on a condition hit while parsing through _renderItems()?

Our current implementation sets "disabled" styling based on a condition within the individual _renderItem() method instead, since our category headers are actually objects within our results list.  Should that method only be called on actual selectable items only?

**Comment by rssteffey** [comment]
@CMTegner 

I've reached a solution that I would love to incorporate into the repo, but if the general consensus is that the functionality doesn't fit then I can just continue with the forked version.

The root of our issue is that we have category headers returned with the rest of the results in our autocomplete list.  These headers need to display regardless of whether any of their category's items were returned, so we can't [cleanly] generate them within our parsing of list content, they need to be part of the items array themselves.

![screen shot 2017-05-25 at 9 53 23 am](https://cloud.githubusercontent.com/assets/4682038/26455496/091da174-4130-11e7-93d8-172553472392.png)

I've added an optional function isItemDisabled() to the Autocomplete component, very much like the existing shouldItemRender() check.  This function currently only gets called within the arrow key events, and is just used to identify and skip non-selectable items.

![arrow_keys_example](https://cloud.githubusercontent.com/assets/4682038/26456687/74a762b0-4133-11e7-843a-7f95ee9db1ed.gif)
If this seems like a good fit for the project, let me know and I'll submit the request.  I'm sure others have run into similar needs at some point. 

_(I am a bit concerned about the function name, so if there are any clearer suggestions let me know.  I think_ "isItemDisabled" _might imply automatic styling and not allowing clicks, which isn't true on my implementation here.  I've only affected the arrow key controls)_
